src/lib.rs

```diff
@@ -1,22 +1,54 @@
-extern crate cfg_if;
-extern crate wasm_bindgen;
+use serde_json::json;
+use worker::*;
 
 mod utils;
 
-use cfg_if::cfg_if;
-use wasm_bindgen::prelude::*;
-
-cfg_if! {
-    // When the `wee_alloc` feature is enabled, use `wee_alloc` as the global
-    // allocator.
-    if #[cfg(feature = "wee_alloc")] {
-        extern crate wee_alloc;
-        #[global_allocator]
-        static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
-    }
+fn log_request(req: &Request) {
+    console_log!(
+        "{:?} - [{}], located at: {:?}, within: {}",
+        Date::now(),
+        req.path(),
+        req.cf().coordinates().unwrap_or_default(),
+        req.cf().region().unwrap_or("unknown region".into())
+    );
 }
 
-#[wasm_bindgen]
-pub fn greet() -> String {
-    "Hello, wasm-worker!".to_string()
+#[event(fetch)]
+pub async fn main(req: Request, env: Env) -> Result<Response> {
+    log_request(&req);
+
+    // Optionally, get more helpful error messages written to the console in the case of a panic.
+    utils::set_panic_hook();
```

Comment:
This is just a side note, not really something that needs to be addressed here. But does this have a significant performance penalty? If not, could this just be baked in, automatically?

Reply:
@Electroid I'm not sure about the exact performance impact, but there is definitely _some_ impact. 

I personally think it improves the dev experience enough to justify it baked in and make it possible to opt-out via features.

Reply:
I'll merge this in as-is, and get the utils::set_panic_hook added in with an opt-out tonight or tomorrow.. just want to test that this template works as expected